lambda/src/firebase-authorizer/firebase-authorizer.js

```diff
@@ -77,8 +78,30 @@ const checkUserAuthorization = async event => {
 
     // Generate AWS authorization policy
     return generatePolicy(user, true, event.methodArn)
-    // There is an authorization token, so validate it.
   }
+
+  // See if the Authorization header is a JSON string containing auth cookie
+  // values.
+  try {
+    const { tabAuthUserTokens, tabAuthUserTokensSig } = JSON.parse(token)
+    const nfaUser = await getUserFromCookies({
+      authCookieValue: tabAuthUserTokens,
+      authCookieValueSig: tabAuthUserTokensSig,
+      includeToken: true,
+    })
+    const user = {
+      uid: nfaUser.id,
+      email: nfaUser.email,
+      email_verified: nfaUser.emailVerified,
+      auth_time: 0, // NFA does not provide an auth time
+    }
+    return generatePolicy(user, true, event.methodArn)
+  } catch (e) {
+    // If not, don't throw. This is expected when provided with a Firebase
```

Comment:
Are there any other non - JSON parse errors we could get there? Might be useful to log if it's not an expected error type.

Reply:
Good suggestion, will update.